targets/disc-configs/lenovo-x1-carbon-disk-config.nix

```diff
@@ -0,0 +1,166 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+_: {
+  disko.devices = {
+    disk.disk1 = {
+      device = "DRIVE_PATH";
+      type = "disk";
+      content = {
+        type = "gpt";
+        partitions = {
+          boot = {
+            name = "boot";
+            size = "1M";
+            type = "EF02";
+          };
+          esp_a = {
+            name = "ESP_A";
+            size = "500M";
+            type = "EF00";
+            content = {
+              type = "filesystem";
+              format = "vfat";
+              mountpoint = "/boot";
+              mountOptions = ["umask=0077"];
+            };
+          };
+          esp_b = {
+            name = "ESP_B";
+            size = "500M";
+            type = "EF00";
+            content = {
+              type = "filesystem";
+              format = "vfat";
+              mountOptions = ["umask=0077"];
+            };
+          };
+          # LVM pool that is going to be encrypted
+          other_1 = {
+            name = "lvm_pv_1";
+            size = "250G";
+            content = {
+              type = "lvm_pv";
+              vg = "pool";
+            };
+          };
+          # LVM pool that is going to be passed to the Storage VM
+          other_2 = {
+            name = "lvm_pv_2";
+            size = "100%";
+            content = {
+              type = "lvm_pv";
+              vg = "vmstore";
+            };
+          };
+        };
+      };
+    };
+    lvm_vg = {
+      pool = {
+        type = "lvm_vg";
+        lvs = {
+          root_a = {
+            size = "50G";
+            content = {
+              type = "filesystem";
+              format = "ext4";
+              mountpoint = "/";
+              mountOptions = [
+                "defaults"
+              ];
+            };
+          };
+          vm_storage_a = {
```

Comment:
How is the vm_storage_a and vm_storage_b used? For system updates it's clear to me that one would apply changes to alternating partitions, but how does this work for vm state?

Reply:
There were some ideas that some VMs can be supplied as pre-built images or so, and the end user will not be able to change the contents of that images. Those VMs will be updated together with the main system and must correspond the specific version of the base system.